R/plots.R

```diff
@@ -146,38 +148,39 @@ generate_individual_outputs <- function(data,
   validate_input_data_generate_individual_outputs(
     data = data,
     matched_prioritized = matched_prioritized,
-    target_type = target_type
+    target_type = target_type,
+    by_group = by_group
   )
 
   # create sub directory for the selected institute
-  dir.create(file.path(output_directory, group_id), showWarnings = FALSE)
+  dir.create(file.path(output_directory, by_group_value), showWarnings = FALSE)
 
   data <- data %>%
     dplyr::filter(
-      group_id == .env$group_id,
-      scenario_source == .env$scenario_source,
-      region == .env$region,
-      sector %in% .env$sector
+      .data[[by_group]] == .env$by_group_value,
+      .data[["scenario_source"]] == .env$scenario_source,
+      .data[["region"]] == .env$region,
+      .data[["sector"]] %in% .env$sector
     )
 
   matched_prioritized <- matched_prioritized %>%
     dplyr::filter(
-      group_id == .env$group_id,
-      sector %in% .env$sector
+      .data[[by_group]] == .env$by_group_value,
+      .data[["sector"]] %in% .env$sector
     )
 
   if (target_type == "tms") {
     # plot tech mix for given sector
     data_techmix <- data %>%
       dplyr::filter(
-        .data$metric %in% c("projected", "corporate_economy", .env$target_scenario),
-        dplyr::between(.data$year, .env$start_year, .env$start_year + .env$time_horizon)
+        .data[["metric"]] %in% c("projected", "corporate_economy", .env$target_scenario),
+        dplyr::between(.data[["year"]], .env$start_year, .env$start_year + .env$time_horizon)
       ) %>%
       dplyr::mutate(
         label = dplyr::case_when(
-          .data$metric == "projected" ~ "Portfolio",
-          .data$metric == "corporate_economy" ~ "Corporate Economy",
-          .data$metric == .env$target_scenario ~ glue::glue("{r2dii.plot::to_title(toupper(.env$scenario))} Scenario")
+          .data[["metric"]] == "projected" ~ "Portfolio",
+          .data[["metric"]] == "corporate_economy" ~ "Corporate Economy",
+          .data[["metric"]] == .env$target_scenario ~ glue::glue("{r2dii.plot::to_title(toupper(.env$scenario))} Scenario")
```

Comment:
nit: with so much of this being these `$` to `[[]]` transformations (which I support, but it's out of context), it's pretty hard to follow what this PR is actually intending to do (enabling `by_group`)

Reply:
I can make a separate PR just transforming $ to [[]] today, should clean this one up significantly